src/account_manager/operations/participation.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_client::{
+    node_api::participation::types::{Event, EventId, EventStatus},
+    node_manager::node::Node,
+    Client,
+};
+
+use crate::account_manager::AccountManager;
+
+impl AccountManager {
+    /// Stores participation information locally and returns the event.
+    ///
+    /// This will NOT store the node url and auth inside the client options.
+    pub async fn register_participation_event(&self, id: EventId, nodes: Vec<Node>) -> crate::Result<Event> {
+        let mut client_builder = Client::builder().with_ignore_node_health();
+        for node in &nodes {
+            client_builder = client_builder.with_node_auth(node.url.as_str(), node.auth.clone())?;
+        }
+        let client = client_builder.finish()?;
+
+        let event_data = client.event(&id).await?;
+
+        let event = Event { id, data: event_data };
+
+        self.storage_manager
+            .lock()
+            .await
+            .insert_participation_event(id, event.clone(), nodes)
+            .await?;
+
+        Ok(event)
+    }
+
+    /// Removes a previously registered participation event from local storage.
+    pub async fn deregister_participation_event(&self, id: EventId) -> crate::Result<()> {
+        self.storage_manager.lock().await.remove_participation_event(id).await?;
+        Ok(())
+    }
+
+    /// Retrieves corresponding information for a participation event from local storage.
+    pub async fn get_participation_event(&self, id: EventId) -> Option<(Event, Vec<Node>)> {
+        self.storage_manager
+            .lock()
+            .await
+            .get_participation_events()
+            .await
+            .unwrap_or_default()
```

Comment:
We're hiding some potentially legit storage issues, I'd return the Result<Option<_>>